Yoast/Sniffs/Yoast/AlternativeFunctionsSniff.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace YoastCS\Yoast\Sniffs\Yoast;
+
+use WordPressCS\WordPress\AbstractFunctionRestrictionsSniff;
+
+/**
+ * Discourages the use of various functions and suggests (Yoast) alternatives.
+ *
+ * @package Yoast\YoastCS
+ * @author  Juliette Reinders Folmer
+ *
+ * @since   1.3.0
+ */
+class AlternativeFunctionsSniff extends AbstractFunctionRestrictionsSniff {
+
+	/**
+	 * Groups of functions to restrict.
+	 *
+	 * @return array
+	 */
+	public function getGroups() {
+		return array(
+			'json_encode' => array(
+				'type'        => 'error',
+				'message'     => 'Detected a call to %s(). Use %s() instead.',
+				'functions'   => array(
+					'json_encode',
+					'wp_json_encode',
+				),
+				'replacement' => 'WPSEO_Utils::format_json_encode',
+			),
+		);
+	}
+
+	/**
+	 * Process a matched token.
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param string $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 *
+	 * @return int|void Integer stack pointer to skip forward or void to continue
+	 *                  normal file processing.
+	 */
+	public function process_matched_token( $stackPtr, $group_name, $matched_content ) {
+
+		$replacement = '';
+		if ( isset( $this->groups[ $group_name ]['replacement'] ) ) {
+			$replacement = $this->groups[ $group_name ]['replacement'];
+		}
+
+		$fixable = true;
+		$error   = $this->groups[ $group_name ]['message'];
```

Comment:
Any reason why this variable is not called `$message`?
As the definition can distinguish between errors and warnings, the message is not always an "error" message.